test/merge-patient-lab-iips-mocha.js

```diff
@@ -5,151 +5,260 @@ var chaiAsPromised = require('chai-as-promised');
+
 var _ = require('underscore');
 var noflo = require('noflo');
+
+var componentFactory = require('../components/merge-patient-lab-iips');
 var getComponent = require('../components/merge-patient-lab-iips').getComponent;
 
+var commonTest = require('./common-test');
+var commonStubs = require('./common-stubs');
+
 describe('merge-patient-lab-iips', function() {
-    it("should have rpf object on nodeInstance", function() {
-        return Promise.resolve(createComponent()).then(function(component){
-            return component.rpf;
-        }).should.be.an('object');
-    });
-    it("should have createVni function", function() {
-        return Promise.resolve(createComponent()).then(function(component){
-            return component.rpf.createVni;
-        }).should.be.a('function');
-    });
-    it("createVni should return an object", function() {
-        return Promise.resolve(createComponent()).then(function(component){
-            return component.rpf.createVni('');
-        }).should.be.an('object');
-    });
-    describe('#createVni', function() {
-        it("should have getInputState function", function() {
-            return Promise.resolve(createComponent()).then(function(component){
-                return component.rpf.createVni('').getInputState;
-            }).should.be.a('function');
+
+    describe('#rpf', function() {
+ 
+        it("should have an rpf object on nodeInstance", function() {
+            return Promise.resolve(componentFactory.getComponent )
+                .then(commonTest.createComponent).then(function(component){
+                    component.rpf.should.exist;
+                    component.rpf.should.be.an('object');
+            });
         });
-        it("should have getOutputState function", function() {
-                return Promise.resolve(createComponent()).then(function(component){
-                return component.rpf.createVni('').getOutputState;
-            }).should.be.a('function');
+
+        it("should have a vni function", function() {
+            return Promise.resolve(componentFactory.getComponent )
+                .then(commonTest.createComponent).then(function(component){
+                    component.rpf.vni.should.exist;
+                    component.rpf.vni.should.be.a('function');
+            });
         });
-        describe('#getInputState', function() {
-            it("should initially return nothing", function() {
-                return Promise.resolve(createComponent()).then(function(component){
-                    return component.rpf.createVni('').getInputState('patient', '', '');
-                }).should.become(undefined);
+
+        describe('#vni', function() {
+
+            it("should return a valid vni object", function() {
+                return Promise.resolve(componentFactory.getComponent )
+                  .then(commonTest.createComponent).then(function(component){
+
+                    var vni = component.rpf.vni('');
```

Comment:
Do we have to require the caller to include an empty string when asking for the IIP vni?